docs/evaluation/tutorials/eval_testing_tutorial.mdx

```diff
@@ -0,0 +1,1313 @@
+      `,
+      content: `ls.test(
+    "should execute code when needed",
+    {
+      inputs: { query: "What was the average return rate for FAANG stock in 2024?" },
+      expected: { answer: 53 },
+    },
+    async ({ inputs: { query }, expected: { answer } }) => {
+      const result = await agent.invoke({
+        messages: [{ role: "user", content: query }],
+      });
+      
+      ls.logOutputs(result);
+      
+      // Grab all the tool calls made by the LLM
+      const toolCalls = result.messages
+        .filter(m => (m as AIMessage).tool_calls)
+        .flatMap(m => (m as AIMessage).tool_calls?.map(tc => tc.name));
+
+      // This will log the number of steps taken by the LLM, which we can track over time to measure performance
+      ls.logFeedback({
+        key: "num_steps",
+        score: result.messages.length - 1, // The first message is the user message
+      });
+
+      // Assert that the tool calls include the "code_tool" function
+      expect(toolCalls).toContain("code_tool");
+
+      // Assert that the answer is within 1 of the expected answer
+      expect(Math.abs(result.structured_response.numeric_answer - answer)).toBeLessThanOrEqual(1);
+    }
+  );`,
+    },
+    {
+      value: "jest",
+      label: "Jest",
+      language: "typescript",
+      caption: `Remember to place each individual test inside a describe block \n
+      \`ls.describe("Agent Tests", () => {
+        // PLACE TESTS Here
+      })\`
+      `,
+      content: `ls.test(
+    "should execute code when needed",
+    {
+      inputs: { query: "What was the average return rate for FAANG stock in 2024?" },
+      expected: { answer: 53 },
+    },
+    async ({ inputs: { query }, expected: { answer } }) => {
+      const result = await agent.invoke({
+        messages: [{ role: "user", content: query }],
+      });
+      
+      ls.logOutputs(result);
+
+      // Grab all the tool calls made by the LLM
+      const toolCalls = result.messages
+        .filter(m => (m as AIMessage).tool_calls)
+        .flatMap(m => (m as AIMessage).tool_calls?.map(tc => tc.name));
+
+      // This will log the number of steps taken by the LLM, which we can track over time to measure performance
+      ls.logFeedback({
+        key: "num_steps",
+        score: result.messages.length - 1, // The first message is the user message
+      });
+
+      // Assert that the tool calls include the "code_tool" function
+      expect(toolCalls).toContain("code_tool");
+
+      // Assert that the answer is within 1 of the expected answer
+      expect(Math.abs(result.structured_response.numeric_answer - answer)).toBeLessThanOrEqual(1);
+    }
+  );`,
+    },
+  ]}
+  groupId="client-language"
+/>
+
+### Test 4: LLM as a judge
+
+For LLM as a judge, we are going to ensure that the agent's answer is grounded in the search results.
+In order to trace the LLM as a judge call separately from our agent, we will use the LangSmith provided `trace_feedback` context manager
```

Comment:
can we link out to SDK docs for`trace_feedback` and wrapEvaluator? i couldnt find